### ISSUE TITLE
For public file viewer have a dropdown "Edit this file in project"

### ISSUE THREAD

**Issue by timothyclemansinsea** [sagemathinc/smc#1428]
One would select the project, SMC would copy the file to the project and open the editable file

**Comment by DrXyzzy** [comment]
Thoughts:
- If the user has multiple projects, maybe offer the 3 (or some small number) most recently visited.
- To shorten dialog, maybe default to saving to `Public` directory in the selected project.
- Borrow code from `Copy` dialog to implement choice of project and doing the copy.

**Comment by DrXyzzy** [comment]
ad #2: default destination dir might be `Downloads` instead of `Public`

**Comment by williamstein** [comment]
>  default destination dir might be Downloads instead of Public

Agreed.  "Public" is bad, since those very files are actually *NOT* public (once copied to the user's project).   Downloads is the standard in most OS's these days.
